import-export-cli/cmd/importAPI.go

```diff
@@ -241,6 +253,122 @@ func mergeAPI(apiDirectory string, environmentParams *params.Environment) error
+				return err
+			}
+			if _, err := api.SetP("", "endpointUTPassword"); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Set the security endpoint parameters when the enabled field is set to true
+// @param envSecurityEndpointParams : Environment security endpoint parameters from api_params.yaml
+// @param api : Parameters from api.yaml
+// @return error
+func setSecurityEndpointsParams(envSecurityEndpointParams *params.SecurityData, api *gabs.Container) error {
+	// Check whether the username, password and type fields have set in api_params.yaml
+	if envSecurityEndpointParams.Username == "" {
+		return errors.New("You have enabled endpoint security but the username is not found in the api_params.yaml")
+	} else if envSecurityEndpointParams.Password == "" {
+		return errors.New("You have enabled endpoint security but the password is not found in the api_params.yaml")
+	} else if envSecurityEndpointParams.Type == "" {
+		return errors.New("You have enabled endpoint security but the type is not found in the api_params.yaml")
+	} else {
+		// Override the username in api.yaml with the value in api_params.yaml
+		if _, err := api.SetP(envSecurityEndpointParams.Username, "endpointUTUsername"); err != nil {
+			return err
+		}
+		// Override the password in api.yaml with the value in api_params.yaml
+		if _, err := api.SetP(envSecurityEndpointParams.Password, "endpointUTPassword"); err != nil {
+			return err
+		}
+		// Set the fields in api.yaml according to the type field in api_params.yaml
+		err := setEndpointSecurityType(envSecurityEndpointParams, api)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Set the fields in api.yaml according to the type field in api_params.yaml
+// @param envSecurityEndpointParams : Environment security endpoint parameters from api_params.yaml
+// @param api : Parameters from api.yaml
+// @return error
+func setEndpointSecurityType(envSecurityEndpointParams *params.SecurityData, api *gabs.Container) error {
+	// Check whether the type is either basic or digest
+	if envSecurityEndpointParams.Type == "digest" {
+		if _, err := api.SetP(true, "endpointAuthDigest"); err != nil {
+			return err
+		}
+	} else if envSecurityEndpointParams.Type == "basic" {
+		if _, err := api.SetP(false, "endpointAuthDigest"); err != nil {
+			return err
+		}
+	} else {
+		// If the type is not either basic or digest, return an error
+		return errors.New("Invalid endpoint security type found in the api_params.yaml. Should be either basic or digest")
+	}
+	return nil
+}
+
 // isEndpointsFieldsValid returns false if either of the two fields: endpoints, loadBalanceEndpoints and failOverEndpoints are defined
 // in api_params.yaml file by the user mistakenly. This will return true , if only one of them is defined.
 func isEndpointsFieldsValid(endpoints *params.EndpointData, loadBalanceEndpoints *params.LoadBalanceEndpointsData, failoverEndpoints *params.FailoverEndpointsData) bool {
```

Comment:
This line can be split. (This comment may be ignored since it's not regarding PR change)